src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/MultiFluidContainer.java

```diff
@@ -0,0 +1,191 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume;
+
+import java.util.AbstractCollection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidVolume;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.api.FluidContainer;
+
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableText;
+
+/**
+ * a fluid container that holds multiple fluid containers in itself.
+ */
+public class MultiFluidContainer extends AbstractCollection<FluidContainer> implements FluidContainer {
+	protected final Collection<FluidContainer> containers;
+
+	/**
+	 * create a multifluid container that wraps the given array.
+	 *
+	 * @param containers the subcontainers for this instance
+	 */
+	public MultiFluidContainer(FluidContainer... containers) {
+		this(Arrays.asList(containers));
+	}
+
+	/**
+	 * create a multifluid container that wraps the given collection.
+	 *
+	 * @param containers the subcontainers for this instance
+	 */
+	public MultiFluidContainer(Collection<FluidContainer> containers) {
+		this.containers = Collections.unmodifiableCollection(containers);
+	}
+
+	@Override
+	public Fraction merge(FluidVolume volume) {
+		return this.combine(volume, FluidContainer::merge);
+	}
+
+	protected Fraction combine(FluidVolume volume, BiFunction<FluidContainer, FluidVolume, Fraction> combiner) {
+		Fraction toMerge = volume.getTotalVolume();
+
+		if (toMerge.equals(Fraction.ZERO)) {
+			return Fraction.ZERO;
+		}
+
+		for (FluidContainer container : this.containers) {
+			Fraction fraction = combiner.apply(container, volume.of(toMerge));
+			this.resync(container);
+			toMerge = toMerge.subtract(fraction);
+			if (toMerge.equals(Fraction.ZERO)) return volume.getTotalVolume();
+		}
+
+		return volume.getTotalVolume().subtract(toMerge);
+	}
+
+	protected void resync(FluidContainer container) {
```

Comment:
Maybe this can be moved upwards and be public?

Reply:
not all con taters need to be synced, and imo, I'd rather not to prevent people from calling it willy-nilly